### ISSUE TITLE
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.

### ISSUE THREAD

**Issue by avpanasyuk** [RandomEngy/VidCoder#956]
### What were you doing when VidCoder crashed?

Trying to start it for the first time

### Crash details

```text
VidCoder 7.3 Beta Installer
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.TypeInitializationException: The type initializer for 'HandBrake.Interop.Interop.HandBrakeEncoderHelpers' threw an exception.
 ---> System.Exception: Please Initialise with HandBrakeUtils.EnsureGlobalInit before using!
   at HandBrake.Interop.Interop.HandBrakeEncoderHelpers..cctor() in C:\git\HandBrake-Fork\win\CS\HandBrake.Interop\Interop\HandBrakeEncoderHelpers.cs:line 39
   --- End of inner exception stack trace ---
   at HandBrake.Interop.Interop.HandBrakeEncoderHelpers.GetContainer(String shortName)
   at VidCoder.Model.PresetStorage.ErrorCheckEncodingProfile(VCProfile profile) in C:\git\VidCoder\VidCoder\Model\PresetStorage.cs:line 251
   at VidCoder.Model.PresetStorage.ErrorCheckPresets(List`1 presets) in C:\git\VidCoder\VidCoder\Model\PresetStorage.cs:line 237
   at VidCoder.Model.PresetStorage.get_UserPresets() in C:\git\VidCoder\VidCoder\Model\PresetStorage.cs:line 54
   at VidCoder.Services.PresetsService..ctor() in C:\git\VidCoder\VidCoder\Services\PresetsService.cs:line 59
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)
   --- End of inner exception stack trace ---
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic, Boolean wrapExceptions)
   at System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture)
   at System.Activator.CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes)
   at System.Activator.CreateInstance(Type type, Object[] args)
   at DryIoc.ReflectionFactory.<>c__DisplayClass21_0.<GetActivator>b__0()
   at DryIoc.Scope.TryGetOrAdd(ImMap`1 items, Int32 id, CreateScopedValue createValue, Int32 disposalOrder)
   at DryIoc.Scope.GetOrAdd(Int32 id, CreateScopedValue createValue, Int32 disposalOrder)
   at DryIoc.ReflectionFactory.CreateServiceExpression(MemberInfo ctorOrMember, Expression factoryExpr, Expression[] paramExprs, Request request)
   at DryIoc.ReflectionFactory.CreateExpressionOrDefault(Request request)
   at DryIoc.Factory.GetExpressionOrDefault(Request request)
   at DryIoc.Factory.GetDelegateOrDefault(Request request)
   at DryIoc.Container.ResolveAndCacheDefaultFactoryDelegate(Type serviceType, IfUnresolved ifUnresolved)
   at DryIoc.Container.DryIoc.IResolver.Resolve(Type serviceType, IfUnresolved ifUnresolved)
   at DryIoc.Resolver.Resolve[TService](IResolver resolver, IfUnresolved ifUnresolved)
   at Microsoft.AnyContainer.DryIoc.DryIocAnyContainer.Resolve[T]()
   at Microsoft.AnyContainer.StaticResolver.Resolve[T]()
   at VidCoder.Services.OutputSizeService..ctor() in C:\git\VidCoder\VidCoder\Services\OutputSizeService.cs:line 16
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)
   --- End of inner exception stack trace ---
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic, Boolean wrapExceptions)
   at System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture)
   at System.Activator.CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes)
   at System.Activator.CreateInstance(Type type, Object[] args)
   at DryIoc.ReflectionFactory.<>c__DisplayClass21_0.<GetActivator>b__0()
   at DryIoc.Scope.TryGetOrAdd(ImMap`1 items, Int32 id, CreateScopedValue createValue, Int32 disposalOrder)
   at DryIoc.Scope.GetOrAdd(Int32 id, CreateScopedValue createValue, Int32 disposalOrder)
   at DryIoc.ReflectionFactory.CreateServiceExpression(MemberInfo ctorOrMember, Expression factoryExpr, Expression[] paramExprs, Request request)
   at DryIoc.ReflectionFactory.CreateExpressionOrDefault(Request request)
   at DryIoc.Factory.GetExpressionOrDefault(Request request)
   at DryIoc.Factory.GetDelegateOrDefault(Request request)
   at DryIoc.Container.ResolveAndCacheDefaultFactoryDelegate(Type serviceType, IfUnresolved ifUnresolved)
   at DryIoc.Container.DryIoc.IResolver.Resolve(Type serviceType, IfUnresolved ifUnresolved)
   at DryIoc.Resolver.Resolve[TService](IResolver resolver, IfUnresolved ifUnresolved)
   at Microsoft.AnyContainer.DryIoc.DryIocAnyContainer.Resolve[T]()
   at Microsoft.AnyContainer.StaticResolver.Resolve[T]()
   at VidCoder.ViewModel.MainViewModel..ctor() in C:\git\VidCoder\VidCoder\ViewModel\MainViewModel.cs:line 306
   at VidCoder.App.OnStartup(StartupEventArgs e) in C:\git\VidCoder\VidCoder\App.xaml.cs:line 170
   at System.Windows.Application.<.ctor>b__1_0(Object unused)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

**Comment by RandomEngy** [comment]
Thanks for the report. I think some code is getting run before initialization, but I don't know what code that would be. Does it crash every time you try to launch?

**Comment by RandomEngy** [comment]
Actually, I think I figured out exactly what was going on. This will be fixed in the next Beta release.

**Comment by RandomEngy** [comment]
Fixed in 7.7 Beta.
